### PR TITLE
Create user's meta dbs correctly

### PR DESCRIPTION
**Pull request by SCdF**
I think this was broken in c9f192ce1eb6e21505dd9a1c69ed648a9b06fa80, but I'm not sure.

Two parts didn't work:
 - db.info() doesn't throw when the DB doesn't exist, it just returns the error
 - We were treating PouchDB instantiation as a promise

### CODE REVIEW

File: api/src/services/user-db.js [hunk 1]
```diff
@@ -63,9 +63,11 @@ module.exports = {
     const dbName = module.exports.getDbName(username);
     return db.exists(dbName).then(found => {
       if (!found) {
-        return db.get(dbName)
-          .then(database => database.put(ddoc))
-          .then(() => setSecurity(dbName, username));
+        const database = db.get(dbName);
+        return database.put(ddoc)
+          .then(result => {
```
Comment (garethbowen) [inline api/src/services/user-db.js]: `result` isn't used, right? It's weird that eslint isn't erroring on this...

File: api/src/db.js [hunk 2]
```diff
@@ -64,7 +64,11 @@ if (UNIT_TEST_ENV) {
   module.exports.exists = name => {
     const db = new PouchDB(getDbUrl(name), { skip_setup: true });
     return db.info()
-      .then(() => true)
+      .then(result => {
+        // In at least PouchDB 7.0.0, info() on a non-existant doc doesn't throw,
```
Comment (garethbowen) [inline api/src/db.js]: I think there's a typo here: 

> non-existant _db_ doesn't throw,

File: api/tests/integration/migrations/utils.js [hunk 3]
```diff
@@ -199,7 +199,7 @@ const _resetDb = () => {
       logger.error('Could not create "medic-test" directly after deleting, pausing and trying again');
       return new Promise(function(resolve, reject) {
         setTimeout(function() {
-          db.get('medic-test')
+          db.exists('medic-test')
```
Comment (garethbowen) [inline api/tests/integration/migrations/utils.js]: I don't think this is correct - it's meant to be trying to create it again. `db.exists` explicitly does not create the db.

File: api/src/services/user-db.js (hunk 1 above)
Comment (SCdF) [reply api/src/services/user-db.js]: Yeah! I noticed this morning that eslint was way too relaxed. If I get time today I'm going to go through our rules and pair them down as much as possible.

File: api/tests/integration/migrations/utils.js (hunk 3 above)
Comment (SCdF) [reply api/tests/integration/migrations/utils.js]: Sorry you're right!

**Comment by garethbowen** [comment]
Merged and backported to `3.4.x`

**Pull request merged**